pkg/products/amqonline/reconciler.go

```diff
@@ -505,6 +506,45 @@ func (r *Reconciler) reconcileConfig(ctx context.Context, serverClient k8sclient
 }
 
 func (r *Reconciler) reconcileBackup(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+
+	postgres, err := croUtil.ReconcilePostgres(ctx, serverClient, defaultInstallationNamespace, r.inst.Spec.Type, postgresTier, r.Config.GetPostgresBackupSecretName(), r.inst.Namespace, r.Config.GetPostgresBackupSecretName(), r.inst.Namespace, func(cr metav1.Object) error {
+		owner.AddIntegreatlyOwnerAnnotations(cr, r.inst)
+		return nil
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to reconcile postgres: %w", err)
+	}
+	// get the secret created by the cloud resources operator
+	croSec := &corev1.Secret{}
+	err = serverClient.Get(ctx, k8sclient.ObjectKey{Name: postgres.Status.SecretRef.Name, Namespace: postgres.Status.SecretRef.Namespace}, croSec)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to get postgres credential secret: %w", err)
+	}
+
+	// create backup secret
+	logrus.Info("Reconciling amq-online postgres backup secret")
+	amqOnlneBackUpSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.Config.GetPostgresBackupSecretName(),
+			Namespace: r.Config.GetNamespace(),
+		},
+		Data: map[string][]byte{},
+	}
+
+	// create or update backup secret
+	_, err = controllerutil.CreateOrUpdate(ctx, serverClient, amqOnlneBackUpSecret, func() error {
+		amqOnlneBackUpSecret.Data["POSTGRES_HOST"] = croSec.Data["host"]
+		amqOnlneBackUpSecret.Data["POSTGRES_USERNAME"] = croSec.Data["username"]
+		amqOnlneBackUpSecret.Data["POSTGRES_PASSWORD"] = croSec.Data["password"]
+		amqOnlneBackUpSecret.Data["POSTGRES_DATABASE"] = croSec.Data["database"]
+		amqOnlneBackUpSecret.Data["POSTGRES_PORT"] = croSec.Data["port"]
+		amqOnlneBackUpSecret.Data["POSTGRES_VERSION"] = []byte("10")
+		return nil
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to create or update %s connection secret: %w", r.Config.GetPostgresBackupSecretName(), err)
+	}
```

Comment:
What do you think of moving this to `reconcileStandardAuthenticationService()` where the postgres is created and making the necessary changes? By doing this there will be a single place for the creation/reconcile of the postGres and fetch / creation of the secrets for the backup seret :thinking: 

Also I think it would be good to check that the postgres is available before fetching the croSecret as otherwise it would throw a panic if it's not present like it's done in codeready

https://github.com/integr8ly/integreatly-operator/blob/54e3cc6b2e6903b2c0763e2e296c424facdd4a6e/pkg/products/codeready/reconciler.go#L199

Reply:
:+1: